### PR TITLE
OOP Java vom 27.04.21

### PR DESCRIPTION
**Pull request by ZeroPointMax**

### CODE REVIEW

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/OOP-CASSELT/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ABWL-LEDER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/DBS-DORENDORF/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/PRO-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWE-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWS-KUSCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/EMB-GUENTHER/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- OOP-CASSELT/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

File: OOP-CASSELT/README.md [hunk 1]
```diff
@@ -183,3 +191,53 @@ if (m instanceof Arbeiter) System.out.println("\t" + ((Arbeiter)m).getLohnsteuer
...
 
 ![Lose Kopplung](assets/lose_kopplung.jpg)<!--width=600px-->
+
+# Ausgewählte Probleme und Aspekte der OOP mit Java
+
+## Liskov und Varianzen
+
+- Wenn eine Klasse erwartet wird, sollte eine Unterklasse davon auch verarbeitet werden können
+- Problemstellung: was darf eine überschreibende Methode ändern, in Bezug auf Rückgabetyp, Typen der formalen Parameter, ggf. auch Typen der Checked Exceptions?
+
```
Comment (Steve2955) [inline OOP-CASSELT/README.md:201]:
```suggestion
```
Die newline würde ich hier glaube ich rausnehmen 🤔

File: OOP-CASSELT/README.md [hunk 2]
```diff
@@ -183,3 +191,53 @@ if (m instanceof Arbeiter) System.out.println("\t" + ((Arbeiter)m).getLohnsteuer
...
+- Implementierung komplexer Datenstrukturen über Zentrale Interfaces wie ``Collection``
+- ``Array``: schneller Zugriff, klassisches Array, aber statische Größe
+- ``ArrayList``: *aufgemotztes* Array, dynamisch in der Größe, wahlfreier Zugriff schneller als in LinkedList, aber Einfügen und Löschen langsamer, da i.d.R ein Umspeichern erforderlich ist
+- ``LinkedList``: doppelt verkettete Liste (siehe Algo Semester 2) dynamischer Größe, passt sich zur Laufzeit an, aber direkter Zugriff langsam, da Liste sequenziell durchlaufen werden muss
+- sowohl ``ArrayList`` als auch ``LinkedList`` erben von ``List``, welches wiederum eine ``Collection`` ist
+
+# Sortieren von Objekten
+
+- 2 Interfaces zur Verfügung gestellt: ``Comparable`` und ``Comparator``
```
Comment (Steve2955) [inline OOP-CASSELT/README.md:239]:
```suggestion
- zwei Interfaces werden zur Verfügung gestellt: ``Comparable`` und ``Comparator``
```

File: OOP-CASSELT/README.md [hunk 3]
```diff
@@ -183,3 +191,53 @@ if (m instanceof Arbeiter) System.out.println("\t" + ((Arbeiter)m).getLohnsteuer
...
+- ``Comparable`` deklariert die Methode public int ``compareTo(Object o)``, welche zu implementieren ist
```
Comment (Steve2955) [inline OOP-CASSELT/README.md:240]:
```suggestion
- ``Comparable`` deklariert die Methode ``public int compareTo(Object o)``, welche zu implementieren ist
```

**Review by Steve2955 — APPROVED** [review]

File: OOP-CASSELT/README.md (hunk 3 above)
Comment (ZeroPointMax) [reply OOP-CASSELT/README.md:240]: wat? ich sehe keinen Unterschied

**Review by nukerxy — APPROVED** [review]

**Pull request merged**